src/index.ts

```diff
@@ -112,18 +112,75 @@ export class IDX {
     return await this._setContent(key, content, options)
   }
 
+  async merge<T extends Record<string, unknown> = Record<string, unknown>>(
+    name: string,
+    content: T,
+    options?: CreateOptions
+  ): Promise<DocID> {
+    const key = this._toIndexKey(name)
+    const existing = await this._getContent<T>(key)
+    const newContent = existing ? { ...existing, ...content } : content
+    return await this._setContent(key, newContent, options)
+  }
+
+  async setAll(
+    contents: Record<string, unknown>,
+    options?: CreateOptions
+  ): Promise<IdentityIndexContent> {
+    const updates = Object.entries(contents).map(async ([name, content]) => {
+      const key = this._toIndexKey(name)
+      const [created, id] = await this._setContentOnly(key, content, options)
+      return [created, key, id]
+    }) as Array<Promise<[boolean, IndexKey, DocID]>>
+    const changes = await Promise.all(updates)
+
+    const newReferences = changes.reduce((acc, [created, key, id]) => {
+      if (created) {
+        acc[key] = id.toUrl('base36')
+      }
+      return acc
+    }, {} as IdentityIndexContent)
+    await this._setReferences(newReferences)
+
+    return newReferences
+  }
+
+  async setDefaults(
+    contents: Record<string, unknown>,
+    options?: CreateOptions
+  ): Promise<IdentityIndexContent> {
+    const index = (await this.getIDXContent()) ?? {}
+
+    const updates = Object.entries(contents)
+      .map(([name, content]) => [this._toIndexKey(name), content])
+      .filter(([key]) => index[key as IndexKey] == null)
+      .map(async ([key, content]) => {
+        const definition = await this.getDefinition(key as IndexKey)
+        const id = await this._createContent(definition, content, options)
+        return { [key as IndexKey]: id.toUrl('base36') }
+      }) as Array<Promise<IdentityIndexContent>>
+    const changes = await Promise.all(updates)
+
+    const newReferences = changes.reduce((acc, keyToID) => {
+      return Object.assign(acc, keyToID)
+    }, {} as IdentityIndexContent)
+    await this._setReferences(newReferences)
+
+    return newReferences
+  }
+
   async remove(name: string): Promise<void> {
     const key = this._toIndexKey(name)
     await this._removeReference(key)
   }
 
   _toIndexKey(name: string): IndexKey {
-    return toCeramicString(this._definitions[name] ?? name)
+    return this._definitions[name] ?? name
   }
 
   // Identity Index APIs
 
-  async getIDXDocID(did?: string): Promise<DocID | null> {
+  async getIDXDocID(did?: string): Promise<string | null> {
```

Comment:
Should this return DocID instance instead of string?

Reply:
It is stored as a string in the DID document so we'd need to explicitly cast it, but it's a bit wasteful for example when doing this because `loadDocument()` in `ceramic-http-client` will need to converts the DocID to string anyways:
```js
const id = await idx.getIDXDocID(did)
const doc = await ceramic.loadDocument(id)
```